### PR TITLE
Move vda-did, vda-did-resolver, vda-web3-clinet

### LINKED ISSUE #216: Move vda-did-resolver from a separate library into the verida-js monorepo
Move vda-did-resolver from a separate library into the verida-js monorepo (vda-did-resolver) and leave the existing did-resolver (we will remove that later).

### PR DESCRIPTION
**Pull request by ITStar10**
Resolves #216  #217

### CODE REVIEW

File: packages/vda-did-resolver/.env.example [hunk 1]
```diff
@@ -0,0 +1,7 @@
+PRIVATE_KEY = "3554d071ecd57a9f0e310209012a23de448c8a9d094303150578a56d0053e7de"
```
Comment (pranavburnwal) [inline packages/vda-did-resolver/.env.example:1]: @ITStar10  There is a private key remove

File: packages/vda-did-resolver/.env.example [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+RPC_TARGET_NET = "RPC_URL_BSC_TESTNET"
+
+RPC_URL_BSC_TESTNET="https://https://data-seed-prebsc-1-s1.binance.org:8545/"
```
Comment (pranavburnwal) [inline packages/vda-did-resolver/.env.example:5]: Can we try to go with Mumbai, please as we are moving to Monorepo?

File: packages/vda-web3-client/src/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable prettier/prettier */
+import { BigNumber } from '@ethersproject/bignumber'
+
+// DID-Registry-Contract address on BSC testnet
+export const DEFAULT_REGISTRY_ADDRESS = '0x2862BC860f55D389bFBd1A37477651bc1642A20B'
```
Comment (pranavburnwal) [inline packages/vda-web3-client/src/helpers.ts:5]: Need to move this to .env

Comment (ITStar10) [reply packages/vda-web3-client/src/helpers.ts:5]: I think it might be better to keep here instead of env.
Because, we can update this default address on our sdk upgrade.

File: packages/vda-web3-client/src/config.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
+/* eslint-disable prettier/prettier */
+import { Signer } from '@ethersproject/abstract-signer'
+
+import { BigNumber } from '@ethersproject/bignumber'
+import { Contract, ContractFactory } from '@ethersproject/contracts'
+import { JsonRpcProvider, Provider } from '@ethersproject/providers'
+import { ethers } from 'ethers'
```
Comment (tahpot) [inline packages/vda-web3-client/src/config.ts:7]: Not used?

File: packages/vda-web3-client/src/config.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+// linked package
+// import DidRegistryContract from 'vda-did-registry'
+import { DEFAULT_REGISTRY_ADDRESS, knownNetworks } from './helpers'
+
+const DidRegistryContract = require('./contract/VeridaDIDRegistry.json')
```
Comment (tahpot) [inline packages/vda-web3-client/src/config.ts:13]: Why is this here? This library is generic for to support any contract.

File: packages/vda-web3-client/src/config.ts [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+        if (conf.rpcUrl) {
+            const chainIdRaw = conf.chainId ? conf.chainId : knownNetworks[conf.name || '']
+            const chainId = chainIdRaw ? BigNumber.from(chainIdRaw).toNumber() : chainIdRaw
+            provider = new JsonRpcProvider(conf.rpcUrl, chainId || 'any')
+        } else {
+        throw new Error(`invalid_config: No web3 provider could be determined for network ${conf.name || conf.chainId}`)
+        }
+    }
+
+    const contract: Contract = ContractFactory.fromSolidity(DidRegistryContract)
```
Comment (tahpot) [inline packages/vda-web3-client/src/config.ts:114]: Wha's the purpose of this?

File: packages/vda-web3-client/src/constants.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+export const veridaContractWhiteList = [
+  '0x2862BC860f55D389bFBd1A37477651bc1642A20B', //DID
+  '0x5c5CA3376b2C82f0322DB9dEE0504D2565080865', //NameRegistry
+  '0x17dFd83eFDD2D0c430E2cA4b01d1df93cDa9960b', //DID
```
Comment (tahpot) [inline packages/vda-web3-client/src/constants.ts:4]: Why is DID here twice?

File: packages/vda-did-resolver/src/configuration.ts [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+import { VeridaSelfTransactionConfig, VeridaMetaTransactionConfig } from '@verida/web3'
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const DIDRegistry = require('./contract/abi.json')
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+require('dotenv').config()
+
+/**
+ * A configuration entry for an ethereum network
+ * It should contain at least one of `name` or `chainId` AND one of `provider`, `web3`, or `rpcUrl`
```
Comment (tahpot) [inline packages/vda-did-resolver/src/configuration.ts:15]: Update comment to show example using `web3`

File: packages/vda-did-resolver/package.json [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+  "description": "Resolve DID documents for ethereum addresses and public keys",
+  "source": "src/index.ts",
+  "main": "lib/index.js",
+  "module": "lib/index.module.js",
+  "unpkg": "lib/index.umd.js",
+  "types": "lib/index.d.ts",
+  "umd:main": "lib/index.umd.js",
+  "repository": {
+    "type": "git",
+    "url": "git@github.com:decentralized-identity/ethr-did-resolver.git"
```
Comment (tahpot) [inline packages/vda-did-resolver/package.json:13]: Update

File: packages/vda-did-resolver/package.json [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+  },
+  "files": [
+    "lib",
+    "src",
+    "LICENSE"
+  ],
+  "author": "Pelle Braendgaard",
```
Comment (tahpot) [inline packages/vda-did-resolver/package.json:20]: Update

File: packages/vda-did-resolver/package.json [hunk 11]
```diff
@@ -0,0 +1,102 @@
+{
+  "name": "vda-did-resolver",
```
Comment (tahpot) [inline packages/vda-did-resolver/package.json:2]: Should be @verida/vda-did-resolver.

Need to update references in other linked packages.

**Review by tahpot — COMMENTED** [review]
@ITStar10 I've added some comments in the code.

I've done a code review, but haven't actually run the code.

Can [VeridaDIDRegistry.json}(packages/vda-web3-client/src/contract/VeridaDIDRegistry.json) be removed as it exists in `did-resolver`?

Need to update `package.json` files with correct attributions and descriptions.

**Comment by ITStar10** [comment]
VeridaDIDRegistry.json need to be shared between all verida js libraries. Once we should publish our DID-Registry contract, it can be resolved. All librqries that require this json file can use the @verida/did-regstry-contract instead of this file.

File: packages/vda-web3-client/src/config.ts (hunk 4 above)
Comment (ITStar10) [reply packages/vda-web3-client/src/config.ts:7]: Used ethers dependency in vda-web3-client.

File: packages/vda-web3-client/src/config.ts (hunk 6 above)
Comment (ITStar10) [reply packages/vda-web3-client/src/config.ts:114]: Contract is used to make transactions in 'web3' mode.

File: packages/vda-web3-client/src/constants.ts (hunk 7 above)
Comment (ITStar10) [reply packages/vda-web3-client/src/constants.ts:4]: That's for test.
The first one was created long ago. Resolving that contract generates a complicated DIDDocument.
For fresh test, I deployed contract again and added address here. Can remove one.
Of course, we should upate this list on deployment.

File: packages/vda-web3-client/src/config.ts (hunk 5 above)
Comment (ITStar10) [reply packages/vda-web3-client/src/config.ts:13]: Removed

**Comment by tahpot** [comment]
This is merging other `blockchain` repos

**Comment by pranavburnwal** [comment]
Global build fix

image

**Comment by ITStar10** [comment]
Merged 'alex/add-vda-did-libraries' branch into `feature/210-decentralized-...` branch.
